tests/stream_test.ts

```diff
@@ -0,0 +1,810 @@
+    const secondPending = await client.xpending_count(
+      key,
+      group,
+      { start: "-", end: "+", count: 10 },
+    );
+    assertEquals(secondPending.length, 3);
+    for (const info of secondPending) {
+      assertEquals(info.owner, "someone");
+      assert(info.lastDeliveredMs > 4);
+      // We called XREADGROUP so it was delivered once
+      // (but not acknowledged yet!)
+      assertEquals(info.timesDelivered, 1);
+    }
+
+    // the output for justIDs will have a different shape
+    const secondClaimedXIds = await client.xclaim(
+      key,
+      { group, consumer: claimingConsumer, minIdleTime, justXId: true },
+      [3000, 0],
+      [3000, 1],
+      [3000, 2],
+    );
+
+    switch (secondClaimedXIds.kind) {
+      case "justxid":
+        assertEquals(
+          secondClaimedXIds.xids,
+          [
+            { unixMs: 3000, seqNo: 0 },
+            { unixMs: 3000, seqNo: 1 },
+            { unixMs: 3000, seqNo: 2 },
+          ],
+        );
+        break;
+
+      default:
+        assert(false);
+    }
+
+    // ACK these messages so we can try XPENDING/XCLAIM
+    // on a new batch
+    await client.xack(key, group, ...secondReply[0].messages.map((m) => m.xid));
+
+    // We'll try one other set of options
+    // for each of XPENDING and XCLAIM
+    await Promise.all(
+      [
+        client.xadd(key, 4000, { "field": "woof", "farm": "chicken" }),
+        client.xadd(key, 5000, { "field": "bop", "farm": "duck" }),
+      ],
+    );
+
+    await client.xreadgroup(
+      [{ key, xid: ">" }],
+      { group, consumer: initialConsumer },
+    );
+
+    // This record won't be included in the filtered
+    // form of XPENDING, below.
+    await client.xadd(key, "*", { "field": "no" });
+    await client.xreadgroup(
+      [{ key, xid: ">" }],
+      { group, consumer: "weird-interloper" },
+    );
+
+    await sleepMillis(5);
+
+    // try another form of xpending: counts
+    // efficiently filtered down to a single consumer.
+    // We expect to see two of the three outstanding
+    // messages here, since one was claimed by
+    // weird-interloper.
+    const thirdPending = await client.xpending_count(
+      key,
+      group,
+      { start: "-", end: "+", count: 10 },
+      "someone",
+    );
+    assertEquals(thirdPending.length, 2);
+    for (const info of thirdPending) {
+      assertEquals(info.owner, "someone");
+      assert(info.lastDeliveredMs > 4);
+      // We called XREADGROUP so it was delivered once
+      // (but not acknowledged yet!)
+      assertEquals(info.timesDelivered, 1);
+    }
+
+    // make sure all the other options can be passed to redis
+    // without some sort of disaster occurring.
+    const thirdClaimed = await client.xclaim(
+      key,
+      {
+        group,
+        consumer: claimingConsumer,
+        minIdleTime,
+        retryCount: 6,
+        force: true,
+      },
+      4000,
+      5000,
+    );
+    switch (thirdClaimed.kind) {
```

Comment:
Can this switch statement be replaced with the following code?

```suggestion
    assert(thirdClaimed.kind === "messages");
    assertEquals(thirdClaimed.messages.length, 2);
    ...
```